src/main/java/cz/centrum/haffner/SimpleTrainingDavid/AppServices/KpisInfoProcessor.java

```diff
@@ -0,0 +1,46 @@
+package cz.centrum.haffner.SimpleTrainingDavid.AppServices;
+
+import cz.centrum.haffner.SimpleTrainingDavid.DataTemplates.KpisInfoData;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import java.util.HashSet;
+import java.util.Set;
+
+@Component
+public class KpisInfoProcessor {
+    private Set differentOriginsSet = new HashSet<Integer>();
+    private Set differentDestinationsSet = new HashSet<Integer>();
+
+    @Autowired
+    private KpisInfoData kpisInfoData;
+
+
+    public void addOneToProcessedFilesNumber() { kpisInfoData.addOneToProcessedFilesNumber(); }
```

Comment:
Když se podíváš na metody v této komponentě a ty v podstatě jen volají úplně stejné metody v jiné komponentě, je na snadě otázka, je vůbec tahle komponenta potřeba?
Co kdyby místo KpisInfoProcessor a KpisInfoData, vznikla jen jedna komponenta: KpisInfoService ?
místo integer fieldů pro differentDestinationsNumber, differentOriginsNumber, by měla rovnou Sety obsahující všechny unikátní zatím zpracovaná čísla?
A processAvgDuration bude taky jednoduchá struktura tam.

Reply:
třídy zmergeovány do již existující KpisInfoData => tř. KpisInfoService už je používána